terraform_modules/states_framework/backend.tf

```diff
@@ -0,0 +1,68 @@
+resource aws_s3_bucket tfstates {
+  bucket              = "nr-coreint-canaries-tfstates"
+  object_lock_enabled = true
+  force_destroy     = true # Allow to destroy all the versions automatically created.
+}
+
+resource aws_s3_bucket_versioning tfstates {
+  bucket = aws_s3_bucket.tfstates.bucket
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_ownership_controls" tfstates {
+  bucket = aws_s3_bucket.tfstates.id
+  rule {
+    object_ownership = "BucketOwnerPreferred"
+  }
+}
+
+resource aws_s3_bucket_acl tfstates {
+  depends_on = [aws_s3_bucket_ownership_controls.tfstates]
+
+  bucket = aws_s3_bucket.tfstates.bucket
+  acl    = "private"
+}
+
+resource aws_s3_bucket_public_access_block tfstates {
+  bucket = aws_s3_bucket.tfstates.id
+
+  block_public_acls   = true
+  block_public_policy = true
+}
+
+resource aws_s3_bucket_object_lock_configuration tfstates {
+  bucket = aws_s3_bucket.tfstates.bucket
+
+  rule {
+    default_retention {
+      mode = "COMPLIANCE"
+      days = 2
```

Comment:
I would set the retention for a month or two.

Files we create are small and add (almost) no cost to New Relic and if we find an issue we are not going to find it after only two days as these are development environments.